src/frontend/context/SettingsContext.js

```diff
@@ -0,0 +1,84 @@
+// @flow
+import * as React from "react";
+import AsyncStorage from "@react-native-community/async-storage";
+
+const STORE_KEY = "@MapeoSettings@36";
+
+export type SettingsContextType = {
+  coordinateSystem?: string,
+};
+
+const initialState = {
+  coordinateSystem: "utm",
+};
+
+const defaultContext = [initialState, () => {}];
+const SettingsContext = React.createContext(defaultContext);
+
+function reducer(state: State, action: Action): State {
+  switch (action.type) {
+    case "set": {
+      return action.value;
+    }
+    case "set_coordinate_system": {
+      try {
+        return { ...state, coordinateSystem: action.value };
+      } catch {
+        return state;
+      }
+    }
+    default:
+      return state;
+  }
+}
+
+const getData = async dispatch => {
+  try {
+    // AsyncStorage.clear()
+    const state = await AsyncStorage.getItem(STORE_KEY);
+    if (state) {
+      const parsedState = JSON.parse(state);
+      dispatch({ type: "set", value: parsedState });
+      return parsedState;
+    } else return initialState;
+  } catch (e) {
+    console.log("Failed to fetch the data from storage");
+    return initialState;
+  }
+};
+
+const saveData = async value => {
+  try {
+    const stringified = JSON.stringify(value);
+    await AsyncStorage.setItem(STORE_KEY, stringified);
+    return value;
+  } catch (e) {
+    console.log("Failed to save the data to the storage");
+  }
+};
+
```

Comment:
There is already a `usePersistedState` hook, so best use that rather than have two different implementations of this.